ml.q

```diff
@@ -258,12 +258,20 @@ cgroup:{[df;X;C]value group imin f2nd[df X] C}
 / matri(X) to the nearest (C)entroid and then uses the (c)entroid (f)unction
 / to update the centroid location.
 lloyd:{[df;cf;X;C]cf X@\: cgroup[df;X;C]}
+/ use (r)esponsibility (f)unction David Mackay's Information Theory..(pg289)
+lloyds:{[df;cf;rf;X;C] cf[rf .ml.f2nd[df X] C;X]} /soft assignment 
 
 kmeans:lloyd[edist2;avg'']      / k-means
 kmedians:lloyd[mdist;med'']     / k-medians
 khmeans:lloyd[edist2;hmean'']   / k harmonic means
 skmeans:lloyd[cosdist;normalize (avg'')::] / spherical k-means
 
+kmeanss:lloyds[edist2;wavg\:/:;dax[=;min]] /k-means using loyd with rf
```

Comment:
i agree that adding this dax here is not very transparent.  how about creating a new function:
```
ismin:dax[=;min]
```

Reply:
the natural place for ismin is next to imin, but dax isn't defined at that point. so I added it after prb, do you prefer it added near first use? (style)